pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -855,16 +856,26 @@ func (r *ReconcileClusterDeployment) statusUpdate(cd *hivev1.ClusterDeployment,
 }
 
 func (r *ReconcileClusterDeployment) resolveInstallerImage(cd *hivev1.ClusterDeployment, imageSet *hivev1.ClusterImageSet, releaseImage string, cdLog log.FieldLogger) (reconcile.Result, error) {
-	if len(cd.Spec.Images.InstallerImage) > 0 {
-		cdLog.WithField("image", cd.Spec.Images.InstallerImage).
-			Debug("setting status.InstallerImage to the value in spec.images.installerImage")
-		cd.Status.InstallerImage = &cd.Spec.Images.InstallerImage
-		return reconcile.Result{}, r.statusUpdate(cd, cdLog)
-	}
+	// we set InstallerImage only if it is not set already to the same image
 	if imageSet != nil && imageSet.Spec.InstallerImage != nil {
-		cd.Status.InstallerImage = imageSet.Spec.InstallerImage
-		cdLog.WithField("imageset", imageSet.Name).Debug("setting status.InstallerImage using imageSet.Spec.InstallerImage")
-		return reconcile.Result{}, r.statusUpdate(cd, cdLog)
+		if cd.Status.InstallerImage != nil && strings.EqualFold(*cd.Status.InstallerImage, *imageSet.Spec.InstallerImage) {
```

Comment:
Are image URI case-insensitive?

Reply:
I would assume so, it just url

Reply:
There are certainly parts of the URI that are case-sensitive, such as the path. I don't see any reason to compare in a case-insensitive fashion here rather than just using the equality operator.

Interestingly, `oc` requires that the URI be all lowercase.
```
$ oc adm release info quay.io/openshift-release-dev/OCP-release:4.2.1
error: repository name must be lowercase
```

Reply:
I was thinking something under these lines https://stackoverflow.com/questions/15641694/are-uris-case-insensitive but I can change

Reply:
My reading of that link supports my contention that the comparison should be done in a case-sensitive fashion.